### ISSUE TITLE
Some partitions get stuck after adding additional consumers to the KEY_SHARED subscriptions

### ISSUE THREAD

**Issue by alexku7** [apache/pulsar#8115]
Hello
We experience the consumers stuck under very strange conditions. It’s still hard to catch . It always happens with KEY_SHARED subscriptions  with multiple consumers.
What we figured out that the trigger for the problem is the adding additional consumers. It can be occurred as part of upgrade ( for example the new upgraded consumers being added and after this the old consumers being removed from the topic) .
Sometimes , because of network glitch, consumer is disconnecting and after few seconds reconnecting. We have 12 consumers on this particular topic but it’s enough that only one consumer will disconnect and reconnect in order to stuck the whole partition for every consumers while other partitions still works fine.
Again, the common trigger is some change with consumers, preferably the adding of new consumer/consumers
 It doesn’t happen always. My estimation that it happens approximately in half of cases.
Of course, no errors on the brokers side expecting the standard messages about connecting new consumers.
The problem is that in order to recover from this situation we need to do one of the following :
Pulsar-admin topics unload topicname
Or we need completely disconnect all consumers and reconnect ( it isn’t recovering if even one consumers remaining connected.. we need to disconnect all of them at once)
Any ideas ?

**Comment by sijie** [comment]
@codelipenghui is working with @alexku7 on this.

**Comment by alexku7** [comment]
Hello
The last status of this issue:

1. It happens in most cases when we add new consumers and remove old or sometime when any current consumer is disconnected and reconnected becasue of small network interruptions.

2. We see some unacked messages (1-5 messages. not more) on the subscription level but zero number of unacked messages on every consumer with 1000 available permits


My guess is that these unacked messages belong to the old consumers that already disconnected and went away forever.

We tried to sen ackTimout flag without success
We tried to set setAllowOutOfOrderDelivery=true to set the relaxed ordering policy still without success
We tried to send ack on every message but still can't avoid this issue


Seems that the broker simply stopped dispatching messages and wait for something forever.

In order to recover from this situation and resume  the dispatching we need to perform the one of the following :
1. to unload the topic or
2. to disconnected the all connected consumer (all of them!) and reconnect again.


Some very strange bug. Seems we need to make the broker to be able to resume dispatching after some timeout despite the fact it has some unacked messages.

Any suggestions ?
